browser/psst/psst_tab_web_contents_observer.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_PSST_PSST_TAB_WEB_CONTENTS_OBSERVER_H_
+#define BRAVE_BROWSER_PSST_PSST_TAB_WEB_CONTENTS_OBSERVER_H_
+
+#include <memory>
+
+#include "base/memory/raw_ptr.h"
+#include "brave/components/psst/browser/core/psst_dialog_delegate.h"
+#include "content/public/browser/web_contents_observer.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+class PrefService;
+
+namespace psst {
+
+class PsstScriptsHandler;
+class PsstRuleRegistry;
+
+FORWARD_DECLARE_TEST(PsstTabWebContentsObserverBrowserTest,
+                     RuleMatchTestScriptTrue);
+
+class PsstTabWebContentsObserver : public content::WebContentsObserver {
+ public:
+  static std::unique_ptr<PsstTabWebContentsObserver> MaybeCreateForWebContents(
+      content::WebContents* contents,
+      Profile* profile,
```

Comment:
icwyu (header or forward decl, but be sure to include the header in the cc file if you forward declare.